zebra/kernel_netlink.c

```diff
@@ -522,7 +522,7 @@ netlink_parse_info (int (*filter) (struct sockaddr_nl *, struct nlmsghdr *,
       if (IS_ZEBRA_DEBUG_KERNEL_MSGDUMP_RECV)
         {
           zlog_debug("%s: << netlink message dump [recv]", __func__);
-          zlog_hexdump(&msg, sizeof(msg));
+          zlog_hexdump(buf, status);
```

Comment:
status gets its value from recvmsg() above which is typically the number of bytes that were received, but if there is an error, it is going to be the error code where status < 0.  I see most errors would have been handled by this point, but (if (errno == EWOULDBLOCK || errno == EAGAIN)) would actually make it to here. zlog_hexdump expect a size of the buffer as a second argument. Are we going to be passing  those two error values in such cases instead of an actual size, that is if there is one to begin with?   

Am I reading this correctly?